evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -364,6 +373,152 @@ where
             }
         }
 
+        // If the allowed memory writes cheatcode is active at this context depth, check to see
+        // if the current opcode can either mutate directly or expand memory. If the opcode at
+        // the current program counter is a match, check if the modified memory lies within the
+        // allowed ranges. If not, revert and fail the test.
+        if let Some(ranges) = self.allowed_mem_writes.get(&data.journaled_state.depth()) {
+            // Helper that expands memory, stores a revert string, and sets the return range to
+            // the revert string's range in memory.
+            fn set_output_buffer(
+                dest_offset: u64,
+                size: u64,
+                interpreter: &mut Interpreter,
+                ranges: &[Range<u64>],
+            ) {
+                let revert_string: Bytes = format!(
```

Comment:
I'd like to extract this to a separate module or end of file, to keep the step fn lean.